the-blue-alliance-ios/View Controllers/Events/EventRankingsTableViewController.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  EventRankingsTableViewController.swift
+//  the-blue-alliance-ios
+//
+//  Created by Zach Orr on 5/14/17.
+//  Copyright © 2017 The Blue Alliance. All rights reserved.
+//
+
+import Foundation
+import TBAKit
+import UIKit
+import CoreData
+
+class EventRankingsTableViewController: TBATableViewController {
+
+    override var persistentContainer: NSPersistentContainer! {
+        didSet {
+            updateDataSource()
+        }
+    }
+    var event: Event!
+    var rankingSelected: ((EventRanking) -> ())?
+    
+    // MARK: - View Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.register(UINib(nibName: String(describing: EventRankingTableViewCell.self), bundle: nil), forCellReuseIdentifier: EventRankingTableViewCell.reuseIdentifier)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        clearsSelectionOnViewWillAppear = splitViewController!.isCollapsed
+    }
+    
+    // MARK: - Refreshing
+    
+    override func refresh() {
+        removeNoDataView()
+        
+        // First things first... refresh all teams for the district, *then* fetch their rankings
+        // Think about if we actually need this - two calls for this is kinda terrible but...
+        var request: URLSessionDataTask?
+        request = TBAEvent.fetchTeams(event.key!, completion: { (teams, error) in
+            if let error = error {
+                self.showErrorAlert(with: "Unable to refresh event rankings - \(error.localizedDescription)")
+            }
+            
+            self.persistentContainer?.performBackgroundTask({ (backgroundContext) in
+                backgroundContext.performAndWait {
+                    teams?.forEach({ (modelTeam) in
+                        _ = Team.insert(with: modelTeam, in: backgroundContext)
+                    })
+                }
+                _ = backgroundContext.saveOrRollback()
+                
+                var rankingsRequest: URLSessionDataTask?
+                rankingsRequest = TBAEvent.fetchRankings(key: self.event.key!, completion: { (rankings, error) in
+                    if let error = error {
+                        self.showErrorAlert(with: "Unable to refresh event rankings - \(error.localizedDescription)")
+                    }
+                    
+                    self.persistentContainer?.performBackgroundTask({ (backgroundContext) in
+                        let backgroundEvent = backgroundContext.object(with: self.event.objectID) as! Event
+                        let realRankings = (rankings?["rankings"] as! [[String: Any]]).map { TBAEventRanking.init(json: $0) }
```

Comment:
Also, this will crash in some cases  - you can have events that don't have ranking information, so you're force casting a null to an array https://www.thebluealliance.com/api/v3/event/2004nj/rankings